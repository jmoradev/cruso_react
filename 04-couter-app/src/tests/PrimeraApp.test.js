import { shallow } from 'enzyme';
import "@testing-library/jest-dom";

import React from 'react';
import PrimeraApp from "../PrimeraApp";


describe('pruebas de primera app', () => {
    /*test('debe mostrar mensaje de hola soy', () => {
        const mensaje = 'hola soy';
        const {getByText} = render(<PrimeraApp saludo={mensaje}/>);
        expect(getByText(mensaje)).toBeInTheDocument();
    })*/
    test('debe mostrar PrimerApp correctamente', () => {
        const saludo = 'soy un saludo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'soy saludo';
        const sub = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={sub} />)

        const textoParrafo = wrapper.find('p').text()
        expect(textoParrafo.trim()).toBe(sub)
    })

})
