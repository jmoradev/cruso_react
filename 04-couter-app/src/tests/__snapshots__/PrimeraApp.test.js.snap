// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de primera app debe mostrar PrimerApp correctamente 1`] = `
<Fragment>
  <h1>
     
    soy un saludo
     
  </h1>
  <p>
     
    Soy un subtitulo
     
  </p>
</Fragment>
`;
